tests/components/remote/test_init.py

```diff
@@ -13,109 +20,128 @@
 
-        assert len(turn_off_calls) == 1
-        call = turn_off_calls[-1]
+async def test_learn_command(hass):
+    """Test learn_command."""
+    learn_command_calls = async_mock_service(hass, DOMAIN, SERVICE_LEARN_COMMAND)
 
-        assert call.domain == remote.DOMAIN
-        assert call.service == SERVICE_TURN_OFF
-        assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    data = {
+        ATTR_ENTITY_ID: ENTITY_ID,
+        ATTR_DEVICE: "test_device",
+        ATTR_COMMAND: ["test_command"],
+        ATTR_COMMAND_TYPE: "rf",
+        ATTR_ALTERNATIVE: True,
+        ATTR_TIMEOUT: 20,
+    }
+    await hass.services.async_call(DOMAIN, SERVICE_LEARN_COMMAND, data)
 
-    def test_send_command(self):
-        """Test send_command."""
-        send_command_calls = mock_service(
-            self.hass, remote.DOMAIN, SERVICE_SEND_COMMAND
-        )
+    await hass.async_block_till_done()
 
-        common.send_command(
-            self.hass,
-            entity_id="entity_id_val",
-            device="test_device",
-            command=["test_command"],
-            num_repeats="4",
-            delay_secs="0.6",
-        )
+    assert len(learn_command_calls) == 1
+    call = learn_command_calls[-1]
 
-        self.hass.block_till_done()
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_LEARN_COMMAND
+    assert call.data[ATTR_ENTITY_ID] == ENTITY_ID
 
-        assert len(send_command_calls) == 1
-        call = send_command_calls[-1]
 
-        assert call.domain == remote.DOMAIN
-        assert call.service == SERVICE_SEND_COMMAND
-        assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+async def test_delete_command(hass):
+    """Test delete_command."""
+    delete_command_calls = async_mock_service(
+        hass, remote.DOMAIN, SERVICE_DELETE_COMMAND
+    )
 
-    def test_learn_command(self):
-        """Test learn_command."""
-        learn_command_calls = mock_service(
-            self.hass, remote.DOMAIN, SERVICE_LEARN_COMMAND
-        )
+    data = {
+        ATTR_ENTITY_ID: ENTITY_ID,
+        ATTR_DEVICE: "test_device",
+        ATTR_COMMAND: ["test_command"],
+    }
 
-        common.learn_command(
-            self.hass,
-            entity_id="entity_id_val",
-            device="test_device",
-            command=["test_command"],
-            alternative=True,
-            timeout=20,
-        )
+    await hass.services.async_call(DOMAIN, SERVICE_DELETE_COMMAND, data)
 
-        self.hass.block_till_done()
+    await hass.async_block_till_done()
 
-        assert len(learn_command_calls) == 1
-        call = learn_command_calls[-1]
+    assert len(delete_command_calls) == 1
+    call = delete_command_calls[-1]
 
-        assert call.domain == remote.DOMAIN
-        assert call.service == SERVICE_LEARN_COMMAND
-        assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.domain == remote.DOMAIN
+    assert call.service == SERVICE_DELETE_COMMAND
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)